packages/headless/src/app/product-listing-engine/product-listing-engine.ts

```diff
@@ -94,6 +117,39 @@ export function buildProductListingEngine(
   };
 }
 
+export function buildProductListingV2Engine(
+    options: ProductListingV2EngineOptions
+): ProductListingV2Engine {
+  const logger = buildLogger(options.loggerOptions);
+  validateConfiguration(options.configuration, logger);
+
+  const productListingClient = createProductListingClient(
+      options.configuration,
+      logger,
+      createSearchAPIClient(options.configuration, logger)
```

Comment:
That seems like an artefact of something that will simply go away eventually (ie: we will never want to rely on the search API at all down the line).

So in other words, by using the "old" product listing client, and having some flag to target v2 or v1 seems to be causing pollution. Which will just get worse IMO.

I think it's better to just do a clean break, create a `commerceServiceClient` (it won't deal with only product listing down the line, after all), and start from there.

Reply:
We need the search api client for facet search, as we currently don't support it on the PLP v2 spec.

In the squad, we talked about creating our own facet search endpoint. I'll reignite the conversation and come back with more info!